Makefile

```diff
@@ -0,0 +1,56 @@
+ALL_SRC := $(shell find . -type f -name '*.go' -not -path "./vendor/*")
+
+GOTEST_OPT=-v -race
+GOTEST=go test
+GOLINT=golint
+GOFMT=gofmt
+GOOS=$(shell go env GOOS)
+
+GIT_SHA=$(shell git rev-parse --short HEAD)
+BUILD_INFO_IMPORT_PATH=github.com/census-instrumentation/opencensus-service/internal/version
+BUILD_INFO=-ldflags "-X $(BUILD_INFO_IMPORT_PATH).GitHash=$(GIT_SHA)"
+
+.DEFAULT_GOAL := default_goal
+
+.PHONY: default_goal
+default_goal: fmt test
+
+.PHONY: clean
+clean:
+	rm -rf bin/ 
+
+.PHONY: test
+test:
+	$(GOTEST) $(GOTEST_OPT) ./...
+
+.PHONY: fmt
+fmt:
+	@FMTOUT=`$(GOFMT) -s -l $(ALL_SRC) 2>&1`; \
+	if [ "$$FMTOUT" ]; then \
+		echo "$(GOFMT) FAILED => gofmt the following files:\n"; \
+		echo "$$FMTOUT\n"; \
+		exit 1; \
+	fi
+
+.PHONY: lint
+lint:
+	@LINTOUT=`$(GOLINT) ./... 2>&1`; \
+	if [ "$$LINTOUT" ]; then \
+		echo "$(GOLINT) FAILED => clean the following lint errors:\n"; \
```

Comment:
Am not sure that we want golint suggestions to fail the build as they are suggestions https://github.com/golang/lint#purpose I personally wouldn't want `mp[key]++` instead of `mp[key] += 1` which is more legible and easier to spot just because of a sweeping `+= 1` should ALWAYS be `++`

Reply:
I will remove golint related stuff from this PR. Let's resolve golint at #141